### DISCUSSION TITLE
Issue with bluetooth sync - BT w/USB to SPDIF.  Seems out by 1/4s and no way to adjust.

### DISCUSSION THREAD

**Discussion by mikwei** [MusicPlayerDaemon/MPD#1967] [General]
Tried setting delay for bluealsa to 0 and delay is less but still not in sync(better), tried negative which negates to option and back to zero..  Any idea's how to fix?  Would be great if mpd.conf each output had a delay option/output.  Manual but better than where I'm at.

Don't mind if other outputs had a say 1000 delay  to sync up audio, not a system wide setting as then video(other outputs)... would be out of sync.

Think it was 100% when I started, but BT was at 100% volume so needed so extra's  to control volume. I could  be mistaken on original as volume so loud.   Seems a delay on BT via alsa but if I can delay others for MPD that would fix.

Not major issue,  as BT will be used when not near stereo, but issue when initial connect is very obvious and annoying if within hearing range of both.  

Anyone have an idea how to resolve?

**Comment by borine** [comment]
MPD does not attempt to synchronize its outputs.

> Tried setting delay for bluealsa to 0

MPD does not use the ALSA "delay" property, so the value reported by bluealsa, or any other ALSA device, is irrelevant as far as the MPD audio output is concerned.

>  if I can delay others for MPD that would fix.

 For audio_outputs of type "alsa", it is possible to add a fixed delay to a PCM device using the ALSA configuration. For an example of sending an audio stream to both `bluealsa` and the local soundcard, with a delay to the soundcard, see https://github.com/arkq/bluez-alsa/wiki/Using-BlueALSA-with-other-ALSA-plugins#multi

However, beware that the ALSA `multi` plugin can suffer underruns if running for long sessions (see for example issue #1874)

To use that example with MPD, add an `audio_output` to your mpd.conf, for example:
```
audio_output {
        name "BT + Local Speakers"
        type "alsa"
        format "48000:16:2"
        device "bt+hw"
        mixer_type "software"
}

```
Incidentally, always use `mixer_type "software"` when using `bluealsa` with MPD.

**Comment by mikwei** [comment]
Thanks for the info. Links were useful, but think I'm stuck.  Not a big
deal as BT mostly used when I'm in other rooms/basement/garage...
